src/agents/parking_agent.py

```diff
@@ -1,33 +1,60 @@
 import spade
+import json
 from spade.agent import Agent
-from spade.behaviour import CyclicBehaviour, OneShotBehaviour
+from spade.behaviour import CyclicBehaviour
 from spade.message import Message
 from spade.template import Template
 
+from constants import DEFAULT_HOST, PARKING_AGENT_MESSAGE_TIMEOUT
+from messages.check_parking import CheckParking
+
 
 class ParkingAgent(Agent):
-    class CheckParkingBehaviour(CyclicBehaviour):
+    def __init__(self, jid, password, x, y, parking_spots):
+        super().__init__(jid, password)
+        self._x = x
+        self._y = y
+        self._parking_spots = parking_spots
+        self._available_parking_spots = [parking_spots for _ in range(24)]  # a list with number of free/available spots at a given hour
+
+    def _prepare_check_parking_spots_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("inform", "parking-availability")
+        return template
+
+    class CheckParkingSpots(CyclicBehaviour):
+        """Behaviour for answering requests for parking spots"""
+
         async def run(self):
-            # @TODO to be implemented
             print("Waiting for incoming messages...")
-            msg = await self.receive(timeout=10)  # Wait for a message for 10 seconds
+            msg = await self.receive(timeout=PARKING_AGENT_MESSAGE_TIMEOUT)
             if msg:
                 print(f"Message received with content: {msg.body}")
-                # Assuming msg.body is requesting the number of available parking spots
-                available_spots = self.get_available_parking_spots()
-                reply = Message(to=msg.sender)  # Create a reply message
-                reply.body = f"Available parking spots: {available_spots}"
+                check_parking_message = CheckParking(**json.loads(msg.body))
```

Comment:
maybe it will be simplier if parking returns only bool that it is available or not ?